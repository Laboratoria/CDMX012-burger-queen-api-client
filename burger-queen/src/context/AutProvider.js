import { createContext, useContext, useState, useEffect } from "react";
import { signInWithEmailAndPassword, onAuthStateChanged, signOut } from 'firebase/auth';
import { auth } from '../firebase-config';

const authContext = createContext();

export const useAuth =() => {
    const context = useContext(authContext);
    if(!context)
        throw new Error('There is not authProvider');
        return context;
};

export function AuthProvider({children}){

    const [user, setUser] = useState(null);

    const login = (email, password) => signInWithEmailAndPassword(auth, email, password);

    const logout = () =>{
        signOut(auth)
    }
    useEffect(()=>{
        const unSuscribe = onAuthStateChanged(auth, (currentUser) =>{
            setUser(currentUser);
        });
        return()=> unSuscribe();
    },[]);

    return (
        <authContext.Provider value={{
            login,
            user,
            logout,
        }} >
            {children}
            </authContext.Provider>
    );

}