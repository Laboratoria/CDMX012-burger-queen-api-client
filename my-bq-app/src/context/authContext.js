import * as React from 'react';
import { createContext, useContext, useEffect, useState} from "react";
import { createUserWithEmailAndPassword, signInWithEmailAndPassword, onAuthStateChanged, signOut } from 'firebase/auth';
import { auth } from '../firebase-config.js';

export const authContext = createContext();

{/*uso de hook para poder llamar los valores sin usar context y createContext desde este mismo hook*/}
export const useAuth = () => {
    const context = useContext(authContext)
    return context
}

export default function AuthProvider ({children}) {

    const [user, setUser] = useState(null);

    const signUp = (email, password) => createUserWithEmailAndPassword(auth, email, password);

    const logIn = (email, password) => signInWithEmailAndPassword(auth, email, password);

    const logOut = () => signOut(auth);

    useEffect(() => {
        onAuthStateChanged(auth, currentUser => {
            setUser(currentUser);
        })
    }, [])
    
    return (
        <authContext.Provider value={{signUp, logIn, user, logOut}}>
            {children}
        </authContext.Provider>
    )
}




