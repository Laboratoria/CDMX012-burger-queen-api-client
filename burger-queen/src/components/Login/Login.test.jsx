import React from 'react';
// import { BrowserRouter as Router } from 'react-router-dom';
// import { create, act } from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import  LoginPage from './Login';
import { AuthProvider } from '../../context/AutProvider';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe(('Test component Login'), () => {
  test('render component Login', () => {

    const {asFragment} = render(< AuthProvider><LoginPage /></AuthProvider>);
    const loginButton = screen.getByText('Login')
    expect(asFragment()).toMatchSnapshot();
    expect(loginButton).toBeDefined();
  });

  test('value when click on button cancel', () => {
    render(< AuthProvider><LoginPage /></AuthProvider> );
    const button = screen.getByText('Login');
    fireEvent.click(button);
    expect(LoginPage.handleSubmit).toHaveBeenCalledTimes(1);
  });
});