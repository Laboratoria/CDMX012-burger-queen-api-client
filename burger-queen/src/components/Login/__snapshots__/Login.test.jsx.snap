// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Login render component Login 1`] = `
<DocumentFragment>
  <div
    class="container-login"
  >
    <form
      class="register-container"
    >
      <img
        alt="logo-icon"
        class="logo-icon"
        src="Logo_Image.gif"
      />
      <h1
        class="title-h1"
      >
         Welcome!
      </h1>
      <h3
        class="title-h3"
      >
        Login to your account
      </h3>
      <div
        class="box-form"
      >
        <img
          alt="email-icon"
          class="icon"
          src="emailIcon.svg"
        />
        <input
          class="input-form"
          name="email"
          placeholder="email@example.com"
          type="email"
        />
      </div>
      <div
        class="box-form"
      >
        <img
          alt="password-icon"
          class="icon"
          src="passwordIcon.svg"
        />
        <input
          class="input-form"
          id="password"
          name="password"
          placeholder="Password..."
          type="password"
        />
      </div>
      <button
        class="btn-form"
        data-testid="btn-login"
        type="button"
      >
         Login
      </button>
      <div
        class="title-h1"
      />
    </form>
  </div>
</DocumentFragment>
`;
